### PR TITLE
feature: add 'help' target to Makefile and make it the default

### PR DESCRIPTION
**Pull request by SataQiu**
### Ⅰ. Describe what this PR did
feature: add 'help' target to Makefile and make it the default

### Ⅳ. Describe how to verify it
make
make help

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -25,127 +27,130 @@ export GO111MODULE=on
...
 
-build-supernode: build-dirs
+build-supernode: build-dirs  ## Build supernode
 	@echo "Begin to build supernode."
 	./hack/build.sh supernode
 .PHONY: build-supernode
 
-install:
-	@echo "Begin to install dfget and dfdaemon and supernode."
+install:  ## Install dfget, dfdaemon and supernode
```
Comment (yunfeiyanggzq) [inline Makefile]: typo dfdaemon ->dfdeamon

File: Makefile [hunk 2]
```diff
@@ -25,127 +27,130 @@ export GO111MODULE=on
...
+	@echo "Begin to install dfget, dfdaemon and supernode."
 	./hack/install.sh install
 .PHONY: install
 
-install-client:
+install-client:  ## Install dfget and dfdaemon
```
Comment (yunfeiyanggzq) [inline Makefile]: ditto

File: Makefile (hunk 1 above)
Comment (SataQiu) [reply Makefile]: Emm... I think `daemon` is right. :man_shrugging:

Comment (yunfeiyanggzq) [reply Makefile]: Emmm.... daemon
```
daemon
ˈdēmən
[名词]. 恶魔, 魔鬼, 妖魔, 鬼子, 恶棍, 精, 妖怪, 鬼蜮
```
deamon
守护进程
```
but in Dragonfly,the most are daemon,you are right,but I think it is spelt wrongly

**Comment by yunfeiyanggzq** [comment]
LGTM

File: Makefile (hunk 1 above)
Comment (SataQiu) [reply Makefile]: Ref: https://translate.google.cn/#view=home&op=translate&sl=en&tl=zh-CN&text=daemon

Comment (yunfeiyanggzq) [reply Makefile]: emmmmmmmmmmmm,I have decided to unstall  my  translation software.

Comment (yunfeiyanggzq) [reply Makefile]: I found you are from buaa? me too!!!!!!

File: Makefile [hunk 3]
```diff
@@ -25,127 +27,130 @@ export GO111MODULE=on
 # Default: use GOPROXY to speed up downloading go mod dependency.
 export GOPROXY=https://goproxy.io
 
-clean:
+help:  ## Display this help information
```
Comment (Starnop) [inline Makefile]: I am curious this section`/^##@/ { printf "\n\033[1m%s\033[0m\n", substr($$0, 5) }`. I think it's useless. 

Maybe you want add `##@` for grouping comments? IMOP, it's useless for now.

Comment (SataQiu) [reply Makefile]: Yes, it is for future grouping.
If you think it doesn't make sense, I can take it out for now. WDYT?

Comment (Starnop) [reply Makefile]: And then could you please add a comment?

Comment (SataQiu) [reply Makefile]: How about "# NB. "##@" will be used to group make goals in the future." ?

Comment (SataQiu) [reply Makefile]: Or just delete `/^##@/ { printf "\n\033[1m%s\033[0m\n", substr($$0, 5) }` ?

Comment (Starnop) [reply Makefile]: In fact, I prefer to remove it. And we can add it if we need in the future. WDYT?

Comment (SataQiu) [reply Makefile]: Already done :)

**Comment by Starnop** [comment]
Well Done. And could you please rebase your code and push it again? THX.

**Comment by SataQiu** [comment]
@Starnop  Rebased!

**Pull request merged**